app/services/bulk_api_import/fhir_condition_importer.rb

```diff
@@ -41,10 +41,19 @@ def build_attributes
     }.compact.with_indifferent_access
   end
 
+  def condition_id
+    @condition_id ||= translate_id(@resource[:subject][:identifier], org_id: @organization_id, ns_prefix: "condition")
+  end
+
   def diagnoses
-    @resource[:code][:coding].each_with_object({hypertension: "no", diabetes: "no"}) do |coding, diagnoses|
-      condition = CONDITION_TRANSLATION[coding[:code]]
-      diagnoses[condition] = "yes"
+    @diagnoses ||= begin
+      prev_htn, prev_dm = MedicalHistory.where(id: condition_id).pluck(:hypertension, :diabetes).first
```

Comment:
[Nitpick] Since we only care for 2 cases, hypertension and diabetes, Wondering if this would be simpler to read if we enumerated the conditions directly?